bin/src/diag.rs

```diff
@@ -149,6 +153,8 @@ pub fn run_diag_read_thread(
                                 let index = qmdl_store.current_entry.expect("DiagDevice had qmdl_writer, but QmdlStore didn't have current entry???");
                                 qmdl_store.update_entry_analysis_size(index, analysis_file_len as usize).await
                                     .expect("failed to update analysis file size");
+                                qmdl_store.update_entry_has_warning(index, analysis_writer.has_warning).await
+                                    .expect("failed to update analysis file has warning");
```

Comment:
imo we shouldn't be storing whether a warning was detected in the QMDL store, but instead should return from `analysis_writer.analyze()` an indication of whether a warning was detected and then update the UI from here.

changing `analyze()`'s return type to be `Result<(usize, bool), std::io::Error>` would be easy enough

Reply:
see above!